aws/rds_subnet_group.go

```diff
@@ -0,0 +1,104 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/cloud-nuke/report"
+	"github.com/gruntwork-io/cloud-nuke/telemetry"
+	"github.com/gruntwork-io/go-commons/errors"
+	commonTelemetry "github.com/gruntwork-io/go-commons/telemetry"
+)
+
+func waitUntilRdsDbSubnetGroupDeleted(svc *rds.RDS, name *string) error {
+	// wait up to 15 minutes
+	for i := 0; i < 90; i++ {
+		_, err := svc.DescribeDBSubnetGroups(&rds.DescribeDBSubnetGroupsInput{DBSubnetGroupName: name})
+		if err != nil {
+			if awsErr, isAwsErr := err.(awserr.Error); isAwsErr && awsErr.Code() == rds.ErrCodeDBSubnetGroupNotFoundFault {
+				return nil
+			}
+
+			return err
+		}
+
+		time.Sleep(10 * time.Second)
+		logging.Logger.Debug("Waiting for RDS Cluster to be deleted")
+	}
+
+	return RdsDeleteError{name: *name}
+}
+
+func getAllRdsDbSubnetGroups(session *session.Session) ([]*string, error) {
+	svc := rds.New(session)
+
+	result, err := svc.DescribeDBSubnetGroups(&rds.DescribeDBSubnetGroupsInput{})
```

Comment:
Can we use the [`DescribeDBSubnetGroupsPages`](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#RDS.DescribeDBSubnetGroupsPages) API call here? Or include logic to handle pagination?

Reply:
SGTM. UPdated.